backend/app/controllers/api/v1/keywords_controller.rb

```diff
@@ -19,7 +19,7 @@ def show
   private
 
   def find_keyword
-    @keyword = Keyword.find_by(id: params[:id])
+    @keyword = Keyword.find_by(id: params[:id], user_id: current_user.id)
```

Comment:
This works 👍 However, adding a condition in each query for the current user would be tedious and easy to forget. A more robust approach is to use authorization policies. Gems like https://github.com/varvet/pundit and https://github.com/palkan/action_policy are the usually preferred implementation options.